docs/Architecture.md

```diff
@@ -0,0 +1,502 @@
+# Babylon Staking Integration Architecture
+
+This is an architectural overview of the various components of Babylon Staking
+integration.
+
+```mermaid
+%%{init: {'theme': 'forest'}}%%
+flowchart TD
+  subgraph Babylon
+  A{{$BTC}} -- Staking --> B(Babylon staking module);
+  B -- $BTC --> C(Local Finality Provider);
+  B -- $BTC Restaking --> D(Zoneconcierge module);
+  end
+
+  D -. IBC .-> E;
+
+  subgraph BSN
+  E(Babylon Contract) -- $BTC Restake --> U(Staking contract);
+  U -- $BTC --> V(BSN Finality Provider);
+  end
+```
+
+You can get a good overview of the whole system flow in the above diagram.
+The design allows Babylon to provide Bitcoin-based security to multiple chains.
+
+## Modules and Contracts
+
+### Babylon Side
+
+On the babylon side, the main module is the Babylon `btcstaking` module, which
+provides the functionality of Babylon Staking integration. It is responsible
+for managing the staking and unstaking of $BTC, as well as providing the
+finality providers with the necessary information to operate.
+There's also a `btcstkconsumer` module, which is responsible for managing the
+staking part that is specific to the Consumer chains.
+The `btcstkconsumer` module interacts with the `zoneconcierge` module, which is
+responsible for managing the restaking of $BTC on the Bitcoin Supercharged
+Networks (BSNs).
+The BSN is a network of Cosmos chains that are integrated with Babylon Staking,
+and receive economic security from Bitcoin.
+
+### BSN Side
+
+On the BSN side, the main contract is the `btc-staking` contract, which is
+responsible for managing the staking and unstaking of $BTC on the BSN. It has
```

Comment:
Current this documentation contains overview of all contracts, and some implementation details of each contract, making it a bit hard to digest.

What do you think we do the following?

- Narrow down the scope of this doc to focus on architecture only
- Create MD file under each contract's folder
- Move relevant part of the `Detailed Interfaces` section to the contract's MD file

We don't need to be perfect in the first version. For now we can just document the most important details, and later we can extend those docs to specs.